infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -143,9 +143,18 @@ public LogReplicationConfig getUpdatedConfig() {
     }
 
     private boolean registryTableHasNewEntries() {
+        // TODO (Chris): the operation of check current tail + generate updated config + update last log tail should be
+        //  atomic operation in multi replication session env.
         StreamAddressSpace currentAddressSpace = runtime.getSequencerView().getStreamAddressSpace(
             new StreamAddressRange(REGISTRY_TABLE_ID, Long.MAX_VALUE, Address.NON_ADDRESS));
         long currentLogTail = currentAddressSpace.getTail();
-        return currentLogTail != lastRegistryTableLogTail;
+
+        // Check if the log tail of registry table moved ahead
+        if (currentLogTail != lastRegistryTableLogTail) {
+            lastRegistryTableLogTail = currentLogTail;
+            return true;
+        }
+
+        return false;
```

Comment:
Not sure why the `lastRegistryTableLogTail` was not getting updated before. Left a TODO for further change to make it synchronized, as we only want to have 1 copy of in-memory registry table entries, while multiple sessions (threads) could query the tail simultaneously